### ISSUE TITLE
Node missing from scene/not position correctly

### ISSUE THREAD

**Issue by HPCR-SD** [godotengine/godot#75167]
### Godot version

4.0.1

### System information

Windows 10

### Issue description

I had a strange issue happen.

FYI i started this project in 4.0 earlier today, I upgraded after the problem happened as I was gonna just use the RC til I saw it went to stable in hopes it would fix this maybe.

I created an animation for something I had centered at the bottom of my game screen.
the animation was to move the position down 32px and bring it back up, upon playing this animation during testing the game, my scene which the animation was in and for, went to 0, 0 on the screen, animated, and then flew up into no where land, never to be seen again during gameplay, except at 0, 0 if it has no script associated, with it, but the script can just be set to extend and it'll disappear with everything else commented out.

This is so weird because you'd think til you play the animation again after resetting it would show again...it shows in the scene for editing, and the node is loaded when you test, but it's at 0, 0 instead of where it should be or no where to be found other then just exisiting in the scene tree

One time just before I caught on to way more details of what is going on I was messing with hiding it in the remote scene tree along with everyother node and it actually appeared and where it should be.

### Steps to reproduce

It must have been the animation...but I can't be sure position is the most basic of animations, this seems like a freak accident that happened and I'm reporting it though, because whether I delete the scene or remove from my other scene this one scene seems to be broken permanently and recreating the scene over again with the same name doesn't fix it if you save in the same folder from what I've found. 

You can get around it, but it's annoying as anything I've ever ran into yet.
In the project I have provided, in case it's not on for you you'll need to have visiable colliders on during debug, or else you will see NOTHING lol I don't have sprites setup yet as I was just working on basics lol.

You will see that I created my paddle in the scene itself and that works, I created the paddle scene in another folder and thankfully THAT works, the res://paddle.tscn jumps to 0, 0 when you start the game. If you delete that scene copy and paste to it the old one (which is what I did for the others I didn't make from scratch because of the polygon collider I have) It will break still if you resave to res://

I'm so sorry if something isn't clear it's been a long night I will do my best tomorrow to clarify anything if needed.

### Minimal reproduction project

[Project.zip](https://github.com/godotengine/godot/files/11024358/Project.zip)

**Comment by HPCR-SD** [comment]
I found that deleting the scene in res:// and then replacing with one I had in test and renaming it to what it was from there, allows it to start working again, but why it broke in the first place, I hope we can figure out so no one else has to deal with this headache.

**Comment by KoBeWi** [comment]
Your `res://paddle` is animating the root node, while `test/paddle` is animating the collider. Animating root node will reset the position of the node to the one you set in animation.

**Comment by HPCR-SD** [comment]
I mean that's fair for that part, but then why did it fly off the tip top corner of the screen from the animation happening at 0, 0 after the animation and then just never show in the scene again properly. It seems after during another test that the scene shows if I have the script attached it's just high enough over 0, 0 that you only see if while animating.

If not script at all which without input doesn't change anything it'll sit at 0, 0 but again the scene position regardless is in the middle of the parent scene and yet it shows in the top left wether 0, 0 or just above so you only see during animation.

**Comment by KoBeWi** [comment]
Ok I don't understand the problem.
What I have to copy where? Give names.

**Comment by HPCR-SD** [comment]
So I may have resolved this, but I'm having a hard time determining if AnimationPlayer is supposed to play [RESET] sometime before or during the ready process? I feel like this morning I just am not searching for info on this in the correct area, or maybe it's not documented/suppose to happen. Basically when I've animated the root node, [RESET] had a position for it of 0, 0 (which doesn't look like was unintended, this was a me issue, but didn't realize this happens/plays when it seems to be playing)

Basically I think if we have a real issue or not comes down to the behavior of AnimationPlayer and the [RESET] animation during loading everything (I'm gonna assume this is intended) as it seems I can't find if this should happen or not, if it should then this can be closed, everything that happened was a mistake on my part/not realizing a behavior of AnimationPlayer. Otherwise I will update with more information tonight

**Comment by HPCR-SD** [comment]
Yeah I just found this just a little earlier too. Upon further information I finally found today I see the bigger picture now. This should be able to be closed. Thank you for pointing me in the right direction.
